pkg/bpv7/extension_block_payload.go

```diff
@@ -41,11 +41,16 @@ func (pb *PayloadBlock) UnmarshalBinary(data []byte) error {
 	return nil
 }
 
-// MarshalJSON writes the binary representation of a PayloadBlock.
+// MarshalJSON creates a json formatted representation of the payload.
 //
+// Since you probably don't want megabytes of encoded data ending up in your logs, large payloads will be truncated.
 // If this type does not implement the json.Marshaler, the CBOR encoding would be returned which might be misleading.
 func (pb *PayloadBlock) MarshalJSON() ([]byte, error) {
-	return json.Marshal(pb.Data())
+	payload := pb.Data()
+	if len(payload) > 100 {
+		payload = payload[:100]
+	}
```

Comment:
Nit Maybe a `const payloadTruncLimit = 100`? 

That said, would it be better to have a `TruncatedJSON()` function that can be used for logging, since `MashalJSON()` kind of sounds like you will get the full payload?

Reply:
I would second @mogren's opinion regarding another method for a truncated output.

Currently this method is used both for JSON logging as well as dtn-tool's output. While one does not want lots of data for the first usecase, it would be very frustrating to be unable to parse a bundle's payload with the second.

I think the way to go would be by creating a custom Logrus formatter for truncated JSON logging. Therefore one could wrap or alter the predefined [JSONFormatter](https://pkg.go.dev/github.com/sirupsen/logrus?utm_source=godoc#JSONFormatter).

Reply:
That sonds like a one way ticket to `reflect`-hell...

Reply:
One could define a second interface to marshal JSON, e.g., `TruncatedJsonMarshaler` next to the `json.Marshaler`. Such a Logrus Formatter could first check, if some type implements a `TruncatedJsonMarshaler`, otherwise pass it to the default `JSONFormatter`.

However, this is probably not the most beautiful solution.